---
title: 'Session 3: Microarray Data Analysis'
author: 
- "S. Rathee, D. Voukantsis, N. Prasad, F. Buffa"
- "Bioinformatics Hub"
date: '`r format(Sys.time(), "Last modified: %d %b %Y")`'
output:
  word_document: null
  pdf_document: default
---

# Note

Please open your R-Studio as an administrator.

# Overview

In this tutorial, we introduce a pipeline to analyse microarray data.

After this tutorial you will be able to:

* Download, load and normalize microarray data.
* Perform PCA on microarray data.
* Find differentially expressed genes from microarray data.
* Design a prediction model.


# Using R Markdown

The tutorial that follows was created using R Markdown. As an exercise in using R Markdown, here, we ask you to save your work in an R Markdown file. By the end of this session, you will have recreated the .Rmd file of this tutorial. You don't have to retype the instruction and comments. Focus only on the R code and add notes that are useful for your own understanding.

To create an R Markdown file:

* Choose **File** -> **New File** -> **R Markdown...** from RStudio menu.
* Save this file with the name **basic_R.Rmd** in your working directory. Don't change the other R Markdown options.
* From R studio menu use **Code** -> **Insert Chunk** to add boxes to save R code.

You may use the R Markdown cheat sheet [https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf](https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf) to further format the output of the R Markdown file.


# R Markdown setup

This is generated by default when we create an R Markdown file. 

```{r setup}

# Change this to match the directory you have stored your scripts and data
root_directory <- "C:/FRANCESCA/Course/Rbio-2019/microarray-new"

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = root_directory)

```

# Required packages

Uncomment and install a package if you didn't already.

```{r Packages}

#source("http://bioconductor.org/biocLite.R") 
# biocLite("affy")
# biocLite("makecdfenv")
# biocLite("GEOquery")
# biocLite("genefilter")
# biocLite("limma")
# biocLite("samr")
# biocLite('dplyr')
# biocLite("caret")
# biocLite("ggfortify")
# biocLite("tidyr")
# biocLite("pheatmap")
# biocLite("RColorBrewer")

```

# Downloading Data

The data we are going to use are cel files (having gene expressions for patients) and clinical data (having response) from a colon cancer study. The data are available at the Geo Database webpage (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41364), but for your convenience may also be found in folder "Rbio-microarray" of this link (https://github.com/bioinformatics-hub/Rbio-microarray).

If you used the Gitshub link, then

* Both files are stored within `session-3.zip`. Click on the file and then the *download* button. 
* Once you have download the file, extract it with winzip or similar program into your current working directory.

If you used the Geo database webpage, then

* Go down the page to find *GSE41364_RAW.tar* (44.7 mb tar file). Download it using http or custom option and extract to pwd.
* Traverse to new webpage by clicking platform *GPL570*. Download *Annotation soft table* to current working directory.
* Create a new clinical data file having sample id and response.

All file  having *.cel* extension includes the gene expression data for 9 samples (patients). The file `clinical-data.txt` includes clinical parameters like response referring to the samples. Annotation file contains information like Gene symbol, entrz id and pathway for probesets used by affymatrix array.

```{r Welcome}

print("Welcome to Session 3")

```

**Question**: If all probesets belongs to some gene?


# Load cdf and cel files

Put all cel files and cdf file in current working directory. Affy package is used to read cel files and create an affy object.

```{r data loading}


```

**Question**: What is the extension of cel files in Affymatrix data?


# Normalization using RMA (Affymetrix microarray suite (MAS) alternate)

Robust Multiarray Averaging (RMA) is a normalisation procedure for microarrays that background corrects, normalises and summarises the probe level information.

This function computes the RMA (Robust Multichip Average) expression measure described in Irizarry et al Biostatistics (2003). Note that this expression measure is given to you in log base 2 scale. This differs from most of the other expression measure methods. Please note that the default background adjustment method was changed during the lead up to the Bioconductor 1.2 release. This means that this function and expresso should give results that directly agree.

```{r RMA}


```

**Question**: Why we use S4 object to store microarray data?


# Load response file to create a phenodata object

Phenodata contains information about patients like their age, response, t-stage, n-stage (example contain response only)

```{r Phenodata}


```

**Question**: How many control samples are available in dataset?


# Upload Gene names for probes


Annotation files are used to provide detailed information about probes. Annotation file contains information like gene name, gene entrz id, pathway etc for a probe.

```{r Annotation file}


```


## PCA (Principal component analysis)

PCA is used to have a clue about signature in a gene set. If first few PCA have high variance portion then there is high possibility of a good signature.

```{r PCA}


```
**Question:** What is variance proportion for first PCA?

# Filtering out probes

We are generally more interested in probes which express well. Therefore, probes having low expression values in 80% of samples should be filter out before analysis. 

```{r Filtering}


```

**Question:** How many probes are filtered out?

## PCA (Principal component analysis) after filtering

PCA is used to have a clue about signature in a gene set. If first few PCA have high variance portion then there is high possibility of a good signature.

```{r PCA2}


```
**Question:** What is variance proportion for first PCA?
## Differentially expressed genes using T-Test and Limma

T-Test is most basic tool to discover differentially expressed genes. Limma is a like a advanced version of T-Test with some improvements over T-Test. In this section, we will discover diffentially expressed probes using t-test and limma to fing common genes for both. 

## T-Test

```{r T-Test}



```

**Question**: How many significant genes for T-Test? 


## Limma


```{r Limma}



```
**Question**: How many significant genes for Limma? 

## SAMR (Significant analysis of Microarrays)

```{r SAMR}


```
**Question**: How many significant genes for SAMR? 

## Vien diagram

Vien diagram can be used to visualize the common probeset for T-Test, Limma and Samr.

```{r Vien Diagram}


```

**Question**: How many significant probes in common for Limma, Samr and T-test? 

## Common gene list with expression values and response

We need commong gene list with expression values and response to make a prediction model.

```{r Common genes}


```
**Question**: What is the name of most significant gene in current colon cancer dataset? 

## HeatMap

Heatmaps are very useful for visualizing the expression of genes and identifying groups of genes with similar expression pattern. There are build-in functions in R for creating heatmaps, but we are going to use the function `pheatmap()` from the pheatmap package, because it is easier to use and creates aesthetically pleasing results.

```{r HeatMap}

```
**Question**: Could we see a separation between A and D samples in heatMap? 

## Prediction model using Support vector machines

Prediction models need two datsets (Training and testing). In this exercise, we will divide our dataset in two sets having 60% and 40 % samples and will use them as training and testing datasets. 

```{r Prediction model}


```